### PR TITLE
Add python support

### PR DESCRIPTION
**Pull request by bogdancondorachi**
@brendt As the title suggests this PR adds Python language support :)

📋 **TODO:**

- [x] add unit tests
- [x] add numbers pattern if #123 approves
- [x] improve argument pattern matching

image

### CODE REVIEW

File: src/Languages/Python/Patterns/PyKeywordPattern.php [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+use Tempest\Highlight\IsPattern;
+use Tempest\Highlight\Pattern;
+use Tempest\Highlight\Tokens\TokenTypeEnum;
+
+final readonly class PyKeywordPattern implements Pattern
+{
+    use IsPattern;
+
+    public function __construct(private array $keywords = [
```
Comment (bogdancondorachi) [inline src/Languages/Python/Patterns/PyKeywordPattern.php:15]: Here I chose to add the keywords as an array directly within the class constructor for simplicity. This offers more efficiency and clarity, especially considering that the list of keywords is unlikely to change frequently.

File: src/Languages/Python/Patterns/PyKeywordPattern.php [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+        'and', 'as', 'assert', 'async', 'await', 'break', 'class', 'continue',
+        'def', 'del', 'elif', 'else', 'except', 'finally', 'for', 'from', 'global',
+        'if', 'import', 'in', 'is', 'lambda', 'nonlocal', 'not', 'or', 'pass',
+        'raise', 'return', 'try', 'while', 'with', 'yield',
+    ])
+    {
+    }
```
Comment (bogdancondorachi) [inline src/Languages/Python/Patterns/PyKeywordPattern.php:22]: Here I chose to add the keywords as an array directly within the class constructor for simplicity. For me this offers more efficiency and clarity, especially considering that the list of keywords is unlikely to change frequently.

File: src/Languages/Python/Patterns/PyBuiltinPattern.php [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+        'complex', 'delattr', 'dict', 'dir', 'divmod', 'enumerate', 'eval', 'exec',
+        'filter', 'float', 'format', 'frozenset', 'getattr', 'globals', 'hasattr',
+        'hash', 'help', 'hex', 'id', 'input', 'int', 'isinstance', 'issubclass', 'iter',
+        'len', 'list', 'locals', 'map', 'max', 'memoryview', 'min', 'next', 'object',
+        'oct', 'open', 'ord', 'pow', 'print', 'property', 'range', 'repr', 'reversed',
+        'round', 'set', 'setattr', 'slice', 'sorted', 'staticmethod', 'str', 'sum',
+        'super', 'tuple', 'type', 'vars', 'zip',
+    ])
+    {
+    }
```
Comment (bogdancondorachi) [inline src/Languages/Python/Patterns/PyBuiltinPattern.php:27]: Same goes for the built-in functions

**Comment by bogdancondorachi** [comment]
Ok @brendt, I will flag this as ready for review. I think it covers well most of the basic stuff to be considered usable.

I'll defiantly continue into more edge cases and additional patterns but for now I'll leave that for a later update.

**Comment by brendt** [comment]
Cool! 

> I'll defiantly continue into more edge cases and additional patterns but for now I'll leave that for a later update.

I agree, this can be finetuned later :) Are you on discord, by any chance? We have a server for Tempest related stuff, it would be nice if I could ping you there — I don't expect you to put in any work you don't want to, but I lack Python knowledge, so it'd be nice to bounce ideas off of someone if we need to further work on this: https://discord.gg/pPhpTGUMPQ

Also, shall I wait until #123 gets merged and you can update this PR accordingly?

**Comment by brendt** [comment]
FYI #123 has been merged :)

**Comment by bogdancondorachi** [comment]
@brendt Added NUMBER and BOOLEAN support. This can be merged now. I'll continue to improve on it when I find time but for now, it should cover enough to be usable.

**Pull request merged**